drivers/modem/hl7800.c

```diff
@@ -2618,8 +2621,10 @@ static bool on_cmd_sock_error_code(struct net_buf **buf, uint16_t len)
 static void sock_notif_cb_work(struct k_work *work)
 {
 	struct hl7800_socket *sock = NULL;
+	struct k_work_delayable *dwork;
 
-	sock = CONTAINER_OF(work, struct hl7800_socket, notif_work);
+	dwork = k_work_delayable_from_work(work);
+	sock = CONTAINER_OF(dwork, struct hl7800_socket, notif_work);
 
 	if (!sock) {
```

Comment:
In what situation would this test succeed?   `CONTAINER_OF` returns an appropriately cast pointer that is offset by some number of bytes from the `work` pointer.  It's not going to be null.

Reply:
Agree, this check isn't needed.